laplace/utils/utils.py

```diff
@@ -219,6 +219,19 @@ def block_diag(blocks):
     return M
 
 
+class SoDSampler(Sampler):
+
+    def __init__(self, N, M, seed: int = 0):
+        np.random.seed(seed)
+        self.indices = torch.tensor(np.random.choice(list(range(N)), M, replace=False))
```

Comment:
Does it make sense to use pure PyTorch here? In this case, `torch.Generator` use be used instead. Globally changing the random seed might have side effect at places far away

Reply:
I believe the only way to do this with torch is using a permutation as `self.indices = torch.randperm(N, generator=rng)[:M]`

Reply:
I see. The main issue for me is that it changes the global rng state. Can you use Numpy's generator?

https://numpy.org/doc/stable/reference/random/generated/numpy.random.Generator.choice.html#numpy.random.Generator.choice